src/main.py

```diff
@@ -358,59 +354,63 @@ async def main() -> None:
-                                )
-                            )
-                            knative_serving_url = hsc_knative_serving_url
-                            in_process_requests_gauge = hsc_in_process_requests_gauge
-                        case 59134:  # HSC upload.py test dataset
-                            hsc_gauge.inc()
-                            fan_out_message_list = (
-                                next_visit_message_updated.add_detectors(
-                                    dataclasses.asdict(next_visit_message_updated),
-                                    hsc_active_detectors_59134,
-                                )
-                            )
-                            knative_serving_url = hsc_knative_serving_url
-                            in_process_requests_gauge = hsc_in_process_requests_gauge
-                        case 59142:  # HSC upload.py test dataset
-                            hsc_gauge.inc()
-                            fan_out_message_list = (
-                                next_visit_message_updated.add_detectors(
-                                    dataclasses.asdict(next_visit_message_updated),
-                                    hsc_active_detectors_59142,
-                                )
-                            )
-                            knative_serving_url = hsc_knative_serving_url
-                            in_process_requests_gauge = hsc_in_process_requests_gauge
-                        case 59150:  # HSC upload.py test dataset
-                            hsc_gauge.inc()
-                            fan_out_message_list = (
-                                next_visit_message_updated.add_detectors(
-                                    dataclasses.asdict(next_visit_message_updated),
-                                    hsc_active_detectors_59150,
-                                )
-                            )
-                            knative_serving_url = hsc_knative_serving_url
-                            in_process_requests_gauge = hsc_in_process_requests_gauge
-                        case 59160:  # HSC upload.py test dataset
-                            hsc_gauge.inc()
-                            fan_out_message_list = (
-                                next_visit_message_updated.add_detectors(
-                                    dataclasses.asdict(next_visit_message_updated),
-                                    hsc_active_detectors_59160,
-                                )
-                            )
-                            knative_serving_url = hsc_knative_serving_url
-                            in_process_requests_gauge = hsc_in_process_requests_gauge
+                        case "HSC":
+                            # HSC has extra active detector configurations just for the
+                            # upload.py test.
+                            match next_visit_message_updated.salIndex:
+                                case 999:  # HSC
```

Comment:
"RC2 full-size dataset"?

Reply:
Good point. To be more precise, this would be any HSC datasets from using `upload_from_repo.py`.  I updated the comment.